### ISSUE TITLE
Research and document DeepSeek V4 ideas applicable to NEAT-AI

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#2526]
## Summary
Produce a single research note that maps each notable DeepSeek V4 technique to NEAT-AI, with a GO / NO-GO recommendation and a one-paragraph rationale per item. This is the foundation document that the experimental sub-issues (#2525) build on.

## What Needs to Be Done
Create `docs/research/deepseek-v4-applicability.md` covering the following V4 ideas, each scored against NEAT-AI's existing architecture (memetic evolution, MCMC acceptance, speciation, discovery cache, backprop, breeding):

1. **GRPO (Group Relative Policy Optimization)** — group-normalised advantage replaces absolute reward; no separate value network. Map to NEAT selection / MCMC acceptance signal.
2. **On-Policy Distillation (OPD)** — single student learns from multiple teacher logits. Map to a new breeding operator.
3. **Muon optimiser** — orthogonalises gradient updates via Newton-Schulz iteration. Map to local backprop pass in `src/propagate/`.
4. **Specialist + generalist distillation pipeline** — independent domain experts trained, then merged via OPD. Map to NEAT speciation (`src/NEAT/Species.ts`, `Genus.ts`).
5. **Manifold-Constrained Hyper-Connections (mHC)** — Birkhoff-polytope projection (Sinkhorn-Knopp) of multi-stream residuals. Map to deep evolved networks with parallel synaptic paths.
6. **Engram conditional memory** — hash-based O(1) n-gram lookup. Compare against existing `src/discovery/SuccessCache.ts` / `FailureCache.ts`.
7. **Hybrid Compressed/Heavily-Compressed Attention (CSA + HCA)** — long-context KV compression. Likely NOT applicable — NEAT-AI has no attention.
8. **FP4/FP8 quantisation-aware training** — memory-efficient training. Likely low value at NEAT scale.

For each item, document:
- One-paragraph technical summary (cite the V4 source).
- The closest existing NEAT-AI module/concept.
- GO / NO-GO recommendation, with the projected experiment.
- Estimated effort (S / M / L) and risk to the UUID-stability and semantic-version invariants in `AGENTS.md`.

## Acceptance Criteria
- [ ] `docs/research/deepseek-v4-applicability.md` exists and covers all eight ideas above.
- [ ] Each idea has a GO or NO-GO recommendation with a one-paragraph rationale.
- [ ] Each GO recommendation links to the corresponding experimental sub-issue.
- [ ] A summary Mermaid diagram shows the V4 idea → NEAT-AI module mapping.
- [ ] No source-code changes; documentation-only PR.
- [ ] `./quality.sh --lint-only` passes (markdown / formatting).

## Dependencies
None — this is the foundation.

## Diagram
```mermaid
flowchart LR
    V4[DeepSeek V4 ideas] --> Doc[applicability.md]
    Doc --> GRPO[GRPO advantage selection]
    Doc --> OPD[On-Policy Distillation breed]
    Doc --> Muon[Muon orthogonalised grads]
    Doc --> Spec[Specialist + distillation]
    Doc --> Engram[Hash-based subnet lookup]
```

## Context
Part of #2525.
Source: https://huggingface.co/deepseek-ai/DeepSeek-V4-Pro/blob/main/DeepSeek_V4.pdf

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/2542 has been created to address this issue.
